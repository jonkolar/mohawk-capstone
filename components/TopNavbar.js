import { useSession, signIn, signOut } from "next-auth/react"

import Link from "./Link";
import LoginButton from "./LoginButton";

import * as React from 'react';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Toolbar from '@mui/material/Toolbar';
import IconButton from '@mui/material/IconButton';
import Typography from '@mui/material/Typography';
import Menu from '@mui/material/Menu';
import MenuIcon from '@mui/icons-material/Menu';
import Container from '@mui/material/Container';
import Avatar from '@mui/material/Avatar';
import Button from '@mui/material/Button';
import Tooltip from '@mui/material/Tooltip';
import MenuItem from '@mui/material/MenuItem';
import AdbIcon from '@mui/icons-material/Adb';
import SearchIcon from '@mui/icons-material/Search';

function ResponsiveAppBar() {
  const { data: session } = useSession()

  const [anchorElNav, setAnchorElNav] = React.useState(null);
  const [anchorElUser, setAnchorElUser] = React.useState(null);

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };
  const handleOpenUserMenu = (event) => {
    setAnchorElUser(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  const handleCloseUserMenu = () => {
    setAnchorElUser(null);
  };

  return (
    <AppBar position="static">
      <Container maxWidth="xl">
        <Toolbar disableGutters>
          <AdbIcon sx={{ display: { xs: 'none', md: 'flex' }, mr: 1 }} />
          <Typography
            variant="h6"
            noWrap
            component="a"
            href="/"
            sx={{
              mr: 2,
              display: { xs: 'none', md: 'flex' },
              fontFamily: 'monospace',
              fontWeight: 700,
              letterSpacing: '.3rem',
              color: 'inherit',
              textDecoration: 'none',
            }}
          >
            LOGO
          </Typography>

          <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
            <IconButton
              size="large"
              aria-label="account of current user"
              aria-controls="menu-appbar"
              aria-haspopup="true"
              onClick={handleOpenNavMenu}
              color="inherit"
            >
            </IconButton>
          </Box>
          <AdbIcon sx={{ display: { xs: 'flex', md: 'none' }, mr: 1 }} />
          <Typography
            variant="h5"
            noWrap
            component="a"
            href=""
            sx={{
              mr: 2,
              display: { xs: 'flex', md: 'none' },
              flexGrow: 1,
              fontFamily: 'monospace',
              fontWeight: 700,
              letterSpacing: '.3rem',
              color: 'inherit',
              textDecoration: 'none',
            }}
          >
            LOGO
          </Typography>
          <Box sx={{ flexGrow: 1, display: 'flex', justifyContent: 'end', marginRight: 1 }}>
              <Link href={"/search"}>
                <SearchIcon />
              </Link>
          </Box>

          <Box sx={{ flexGrow: 0 }}>
            { session ?
            <>
              <Tooltip title="Open settings">
                <IconButton onClick={handleOpenUserMenu} sx={{ p: 0 }}>
                  <Avatar alt="Avatar" src={session.user.image} />
                </IconButton>
              </Tooltip>
              <Menu
                sx={{ mt: '45px' }}
                id="menu-appbar"
                anchorEl={anchorElUser}
                anchorOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                keepMounted
                transformOrigin={{
                  vertical: 'top',
                  horizontal: 'right',
                }}
                open={Boolean(anchorElUser)}
                onClose={handleCloseUserMenu}
              >
              {!session.user.username ? 
                <MenuItem onClick={handleCloseUserMenu}>
                  <Link href={"/account/choose-username"}>
                    <Typography textAlign="center">Choose Username</Typography>
                  </Link>
                </MenuItem>
                :
                <MenuItem
                  sx={{"&:hover": {backgroundColor: 'transparent', cursor: 'default' }}}
                >
                    <Typography textAlign="center">Signed in: {session.user.username}</Typography>
                </MenuItem>
              }
                <MenuItem onClick={handleCloseUserMenu}>
                  <Link href={"/users/" + session.user.username}>
                    <Typography textAlign="center">Profile</Typography>
                  </Link>
                </MenuItem>
                <MenuItem onClick={handleCloseUserMenu}>
                  <Link href={"/account/invites"}>
                    <Typography textAlign="center">Team Invites</Typography>
                  </Link>
                </MenuItem>
                <MenuItem onClick={handleCloseUserMenu}>
                  <Link href={"/account/teams"}>
                    <Typography textAlign="center">Teams</Typography>
                  </Link>
                </MenuItem>
                <MenuItem onClick={() => signOut()}>
                    <Typography textAlign="center">Sign Out</Typography>
                </MenuItem>
              </Menu>
            </>
            :
            <Button variant="contained" onClick={() => signIn('discord')}>Sign In</Button>
          }
          </Box>
        </Toolbar>
      </Container>
    </AppBar>
  );
}
export default ResponsiveAppBar;