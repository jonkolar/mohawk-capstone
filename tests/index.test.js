import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";

import TopNavbar from "@/components/TopNavbar";
import { SessionProvider } from "next-auth/react";

jest.mock("next-auth/react", () => {
    const originalModule = jest.requireActual('next-auth/react');
    const mockSession = {
      expires: new Date(Date.now() + 2 * 86400).toISOString(),
      user: { username: "admin" }
    };
    return {
      __esModule: true,
      ...originalModule,
      useSession: jest.fn(() => {
        return { data: mockSession, status: 'authenticated' }  // return type is [] in v3 but changed to {} in v4
      }),   
    };
});

describe("Navbar Component", () => {

    it('Show Log Out when has session',
      async () => {
        const { container } = render(<TopNavbar/>);
  
        expect(screen.getByText("Sign Out")).toBeInTheDocument();
      })
  
  })