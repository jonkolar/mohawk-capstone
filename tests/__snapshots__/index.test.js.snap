// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Show Log Out when has session 1`] = `
<div>
  <h1>
    Home
  </h1>
</div>
`;
